pkgs/tools/misc/rtx/default.nix

```
@@ -0,0 +1,72 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, installShellFiles
+, stdenvNoCC
+, perl
+, coreutils
+, bash
+, direnv
+, Security
+}:
+
+let
+  owner = "jdxcode";
+  pname = "rtx";
+  version = "1.29.7";
+in
+rustPlatform.buildRustPackage rec {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-w61ik1Tne7cO1Aez4aem3/UQfWjoCTnz+vR1YaaHDhY=";
+  };
+
+  cargoSha256 = "sha256-PGq/LSM8bsNWRVDLLhaiEgaXIN+957HnF0SUx6M7Wus=";
+
+  # perl because: https://docs.rs/openssl/0.10.45/openssl/#vendored
+  nativeBuildInputs = [ perl installShellFiles ];
+  buildInputs = lib.optionals stdenvNoCC.isDarwin [ Security ];
+
+  # see for reference: https://github.com/jdxcode/rtx/blob/main/default.nix
+  prePatch = ''
+    substituteInPlace ./test/data/plugins/**/bin/* \
+      --replace '#!/usr/bin/env bash' '#!${bash}/bin/bash'
+    substituteInPlace ./src/fake_asdf.rs ./src/cli/reshim.rs \
+      --replace '#!/bin/sh' '#!${bash}/bin/sh'
+    substituteInPlace ./src/env_diff.rs \
+      --replace '"bash"' '"${bash}/bin/bash"'
+    substituteInPlace ./src/cli/direnv/exec.rs \
+      --replace '"env"' '"${coreutils}/bin/env"' \
+      --replace 'cmd!("direnv"' 'cmd!("${direnv}/bin/direnv"'
+  '';
+
+  checkFlags = [
+    # Skip the test_plugin_list_urls as it uses the .git folder, which
+    # is excluded by default from Nix.
+    # see for reference: https://github.com/jdxcode/rtx/blob/main/default.nix
+    "--skip=cli::plugins::ls::tests::test_plugin_list_urls"
+  ];
+  cargoTestFlags = [ "--all-features" ];
+  # some tests access the same folders, don't test in parallel to avoid race conditions
+  dontUseCargoParallelTests = true;
+
+  postInstall = ''
+    installManPage ./man/man1/rtx.1
+    installShellCompletion \
+      ./completions/rtx.{bash,fish} \
+      --zsh ./completions/_rtx
+  '';
+
+  meta = rec {
+    homepage = "https://github.com/${owner}/${pname}";
+    description = "Polyglot runtime manager (asdf rust clone)";
+    changelog = "${homepage}/releases/tag/v${version}";
```

```suggestion
    changelog = "https://github.com/jdxcode/rtx/releases/tag/v${version}";
```
lets be safe that a change in homepage is not going to break this